Use tree-sitter Java parser

aryx opened the pull request.

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: started filled in
with code to convert to ast_java.ml

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml:

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml:

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml:

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml:

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: use canon_var

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: using Ast_java.decls

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: v1

aryx committed:
* pfff: use ast_java_tree_sitter branch

aryx commented:
This should help https://github.com/returntocorp/semgrep/issues/1189

aryx commented:
This closes https://github.com/returntocorp/semgrep/issues/1257

mjambon reviewed (approved):
Looks like you're doing ok :-)

My upcoming changes due to factorization should not break things.

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: remove @@warning and fix warnings

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: progress

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: use intersection types

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: enum constant

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml: final todo()!

aryx committed:
* semgrep-core/parsing/Parse_java_tree_sitter.ml:

aryx committed:
* pfff: use latest of branch

aryx committed:
* semgrep-core/tests/PARSING/annot_interface.java:

This show https://github.com/returntocorp/semgrep/issues/1303 now works

test plan:
+ /home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -dump_ast annot_interface.java

aryx committed:
* semgrep-core/tests/PARSING/annot_interface2.java:

This show https://github.com/returntocorp/semgrep/issues/1198 now works

Test plan:
+ /home/pad/github/semgrep/semgrep-core/_build/default/bin/Main.exe -dump_ast annot_interface2.java
no parse error

aryx committed:
* semgrep-core/tests/PARSING/annotGeneric.java:

https://github.com/returntocorp/semgrep/issues/1197 now works

aryx committed:
* semgrep-core/tests/PARSING/enumArgsAndBody.java:

This closes https://github.com/returntocorp/semgrep/issues/1196

aryx committed:
* semgrep-core/tests/PARSING/generics2.java: this now works

This closes https://github.com/returntocorp/semgrep/issues/1195

aryx committed:
* semgrep-core/tests/PARSING/generics3.java: it works now

this closes https://github.com/returntocorp/semgrep/issues/1194

aryx committed:
* semgrep-core/tests/PARSING/generics1.java: This works now

This closes https://github.com/returntocorp/semgrep/issues/1184

aryx committed:
* semgrep-core/tests/PARSING/generics0.java: it works now

this closes https://github.com/returntocorp/semgrep/issues/1048

aryx committed:
* semgrep-core/tests/PARSING/Ghidra.java: it works now

This closes https://github.com/returntocorp/semgrep/issues/993

aryx committed:
* semgrep-core/tests/PARSING/Ghidra2.java: it works now

this closes https://github.com/returntocorp/semgrep/issues/992

aryx committed:
* semgrep-core/tests/PARSING/generics4.java: it works now

this closes https://github.com/returntocorp/semgrep/issues/990

aryx committed:
* pfff: use latest

aryx committed:
* semgrep-core/tests/PARSING/annotStmt.java: more

brendongo committed:
Bump pfff

brendongo committed:
Merge branch 'develop' into tree_sitter_java

aryx committed:
* pfff: bump more
